packages/katello/katello/hostname-change.rb

```diff
@@ -239,6 +291,122 @@ def setup_opt_parser
       @opt_parser.parse!
     end
 
+    def dns_managed?
+      @scenario_answers['foreman_proxy']['dns'] &&
+        @scenario_answers['foreman_proxy']['dns_managed'] &&
+        @scenario_answers['foreman_proxy']['dns_provider'] == 'nsupdate'
+    end
+
+    def should_update_dns?
+      dns_managed? && !@options[:skip_dns]
+    end
+
+    def query_dns_server(fqdn, zone, nameserver_ip)
+      query_dns = Resolv::DNS.new(nameserver: [nameserver_ip], search: [], ndots: 1)
+      a_record = query_dns.getresource(fqdn, Resolv::DNS::Resource::IN::A)
+      ip = a_record.address.to_s if a_record
+      soa_record = query_dns.getresource(zone, Resolv::DNS::Resource::IN::SOA)
+      serial = soa_record.serial if soa_record
+      {
+        ip: ip,
+        serial: serial
+      }
+    end
+
+    def new_dns_values
+      new_vals = OpenStruct.new
+
+      new_vals.dns_server = @scenario_answers['foreman_proxy']['dns_server']
+      new_vals.zone = @scenario_answers['foreman_proxy']['dns_zone']
+      new_vals.key_file = @scenario_answers['foreman_proxy']['keyfile']
+
+      new_vals.reverse_zones = [@scenario_answers['foreman_proxy']['dns_reverse']].flatten
+      ip_serial_data = query_dns_server(@old_hostname, new_vals.zone, new_vals.dns_server)
+      new_vals.ip = ip_serial_data[:ip]
```

Comment:
With the new update_zone this is no longer needed.

Reply:
@ekohl I wanted to leave a couple of those values in because the `precheck` method also calls this.  If we get to the point in the script where we're updating DNS and it fails, the change-hostname script fails quite badly and leaves the system in a state where it's not easy to just run again.  So we query the DNS resolver for values before the process starts.

Reply:
@ekohl Found a better way to do the precheck; updating soon

Reply:
Updated to use `assembled_nsupdate_command` in the precheck.  Thanks @jturel for the suggestion :)